pages/learn/deploy/deployment.md

```diff
@@ -1,52 +1,250 @@
+
+In order to allow options like emulation and nocache, the `git push` workflow uses specifically named remote branches (see next section) however this has the limitation that it is not possible to invalidate the cache of a emulated build pushed with `git push`.
+
+### Additional Options
+
+Like `balena push` the `git push` workflow also allows one to trigger a build that invalidates the docker layer cache and builds from scratch. This can be achieved by pushing to a special branch called `balena-nocache` as shown in the example below.
+```
+$ git push balena master:balena-nocache
+```
+Similarly one can also trigger a [qemu](https://www.qemu.org/) build on the [balenaCloud](#the-balenacloud-build-server) build server by pushing to the `balena-emulated` remote branch as shown below:
+```
+$ git push balena master:balena-emulated 
+```
+
+## Project Structure
+
+When deploying a balena project the build system will endeavour to build the most appropriate release for a specific set of devices. The following section will discuss some of the mechanisms you can use to control the type of builds that are produced. 
+
+### Project Resolutions
+
+All the deployment methods will always try determine the project type based on the following project resolution ordering.
+
+**Project Resolution Ordering**
+- docker-compose.yml
+- Dockerfile.<device-type>
+- Dockerfile.<arch>
+- Dockerfile.template
+- Dockerfile
+- package.json
+
+This resolution mechanism looks at the files in the root of the directory you are deploying. If it finds a `docker-compose.yml` file it will ignore all the other types and build a multicontainer release based on the service specification in the `docker-compose.yml` .
+
+If there is no `docker-compose.yml` specified, the resolution system will assume a single container deployment and will build based on a `Dockerfile.*` file. These dockerfiles can have extensions of `.<device-type>`, `<.arch>` or `.template` and the build system will use the most appropriate file based on the targeted device or application. This is best described with an example. 
+
+In our example at the root of our project repo we have the following `Dockerfile.*` files:
+```
+project: $ tree -a
+.
+├── Dockerfile.raspberrypi3
+├── Dockerfile.i386
+└── Dockerfile
+```
+
+When we push this project to an application which has its default device type selected as `Raspberry Pi 3`, the build system will use the device type specific `Dockerfile.raspberrypi3` file to build from. If we instead pushed this to an `Intel Edison` application, the build would use the `Dockerfile.i386` file. When pushing to any other device type, the regular `Dockerfile` would be used to perform the build. This type of project selection will also work in service folders of multicontainer deployments, you can see an example of that in our [Getting started with multicontainer project](https://github.com/balena-io-projects/multicontainer-getting-started/tree/master/haproxy). 
```

Comment:
change comma to semicolon in "This type of project selection will also work in service folders of multicontainer deployments, you can see an example.."